packages/scalecube-discovery/src/Discovery/Discovery.ts

```diff
@@ -0,0 +1,25 @@
+import { Discovery as DiscoveryInterface, DiscoveryConnect, DiscoveryCreate } from '../api/public';
+import { getSeed, notifyAllListeners, removeFromCluster, addToCluster } from './DiscoveryActions';
+import { ReplaySubject } from 'rxjs';
+import { Endpoint } from '@scalecube/scalecube-microservice/src/api/public';
+
+export const Discovery: DiscoveryInterface = Object.freeze({
+  create({ address, endPoints, seedAddress }: DiscoveryConnect): DiscoveryCreate {
+    let seed = getSeed({ seedAddress });
+    const subjectNotifier = new ReplaySubject<Endpoint[]>(1);
+
+    seed = addToCluster({ seed, address, endPoints, subjectNotifier });
+    notifyAllListeners({ seed });
+
+    return Object.freeze({
+      end: () => {
+        seed = removeFromCluster({ seed, address });
+        subjectNotifier && subjectNotifier.unsubscribe();
```

Comment:
As I understand, you can call **unsubscribe()** on the subscription, not on the stream itself

Reply:
i'll change it to subjectNotifier .complete()